pkgs/racket-doc/scribblings/reference/filesystem.scrbl

```diff
@@ -512,6 +512,70 @@ In all modes, the @exnraise[exn:fail:filesystem] on error (e.g., if no
 such file exists).}
 
 
+@defproc[(file-or-directory-stat [path path-string?]
+                                 [as-link? boolean? #f])
+         (hash/c symbol? any/c)]{
+
+@index['("inode")]{Returns} a hash with the following keys and values:
+
+@itemlist[
+ @item{@racketvalfont{'device-id} : device id}
```

Comment:
For each of these, I think the documentation should describe the representation, such as "an exact nonnegative integer".

Reply:
Instead of `#racketvalfont{'device-id}`, `@racket['device-id]` is better. In this case, `indexed-racket` is probably best. In general, prefer `racket` to `racketvalfont` until you need to control the textual representation, such as for showing octal.

Reply:
Yes, _currently_ all hash entries are non-negative integers. At some point there was a plan to include a file type field, whose values would be symbols, and I don't know if we'll add other non-integer fields in the future. Therefore I'm not sure what's better: Be more precise regarding the current implementation or make a potentially wrong "promise" which may cause problems for client code if it relies on the non-negative-integer assumption for the future.

Reply:
> For each of these, I think the documentation should describe the representation, such as "an exact nonnegative integer".

Can you give an example where you would put this information? Anyway, I'm not sure it's needed because someone using the stat function has to know what the fields mean and what their possible values are. On the other hand, the Windows values 0 for uid/gid are a bit special and possibly worth documenting.

Reply:
> For each of these, I think the documentation should describe the representation, such as "an exact nonnegative integer".

Somewhat related: The contract of the function says about the hash: `(hash/c symbol? any/c)`. _Currently_ all hash values are non-negative integers, but I'm using `any/c` to avoid promising this for future enhancements of the function. Actually, at one point we discussed including a `'type` key whose value would be a symbol like `'directory`. There's still (commented-out) documentation on this.